web/packages/teleport/src/Login/Login.test.tsx

```diff
@@ -114,41 +115,65 @@ test('login with private key policy enabled through role setting', async () => {
   expect(screen.getByText(/login disabled/i)).toBeInTheDocument();
 });
 
-test('show motd only if motd is set', async () => {
-  // default login form
-  const { unmount } = render(<Login />);
-  expect(screen.getByPlaceholderText(/username/i)).toBeInTheDocument();
-  expect(
-    screen.queryByText('Welcome to cluster, your activity will be recorded.')
-  ).not.toBeInTheDocument();
-  unmount();
-
-  // now set motd
-  jest
-    .spyOn(cfg, 'getMotd')
-    .mockImplementation(
-      () => 'Welcome to cluster, your activity will be recorded.'
-    );
-
-  render(<Login />);
+describe('test MOTD', () => {
+  test('show motd only if motd is set', async () => {
+    // default login form
+    const { unmount } = render(<Login />);
+    expect(screen.getByPlaceholderText(/username/i)).toBeInTheDocument();
+    expect(
+      screen.queryByText('Welcome to cluster, your activity will be recorded.')
+    ).not.toBeInTheDocument();
+    unmount();
+
+    // now set motd
+    jest
+      .spyOn(cfg, 'getMotd')
+      .mockImplementation(
+        () => 'Welcome to cluster, your activity will be recorded.'
+      );
+
+    render(<Login />);
+
+    expect(
+      screen.getByText('Welcome to cluster, your activity will be recorded.')
+    ).toBeInTheDocument();
+    expect(screen.queryByPlaceholderText(/username/i)).not.toBeInTheDocument();
+  });
 
-  expect(
-    screen.getByText('Welcome to cluster, your activity will be recorded.')
-  ).toBeInTheDocument();
-  expect(screen.queryByPlaceholderText(/username/i)).not.toBeInTheDocument();
-});
+  test('show login form after modt acknowledge', async () => {
+    jest
+      .spyOn(cfg, 'getMotd')
+      .mockImplementation(
+        () => 'Welcome to cluster, your activity will be recorded.'
+      );
+    render(<Login />);
+    expect(
+      screen.getByText('Welcome to cluster, your activity will be recorded.')
+    ).toBeInTheDocument();
+
+    fireEvent.click(screen.getByText('Acknowledge'));
+    expect(screen.getByPlaceholderText(/username/i)).toBeInTheDocument();
+  });
 
-test('show login form after modt acknowledge', async () => {
-  jest
-    .spyOn(cfg, 'getMotd')
-    .mockImplementation(
-      () => 'Welcome to cluster, your activity will be recorded.'
+  test('skip motd if login initiated from headless auth', async () => {
+    jest
+      .spyOn(cfg, 'getMotd')
+      .mockImplementation(
+        () => 'Welcome to cluster, your activity will be recorded.'
+      );
+
+    const url =
+      'https://teleport.example.com/web/login?redirect_uri=https://teleport.example.com/web/headless/5c5c1f73-ac5c-52ee-bc9e-0353094dcb4a';
+    render(
+      <MemoryRouter>
```

Comment:
Unfortunately, this test doesn't really work. `Route` component doesn't set the active route to the given path, but instead it tells the `Router`: "render my children when the active location equals my `path`".
So, this test passes because nothing is rendered :)
What's more, in `beforeEach` we mock the `history.getRedirectParam` so it always returns `/`.  

IMO, the best way to test this it to mock this function in our test:
```ts
jest.spyOn(history, 'getRedirectParam').mockReturnValue('https://teleport.example.com/web/headless/5c5c1f73-ac5c-52ee-bc9e-0353094dcb4a');
```
and remove `MemoryRouter` and `Route`.